### PR TITLE
[AAG-7] Create the intent layer

### PR DESCRIPTION
**Pull request by cnavarrete101**
Creation of the use cases to handle the change of the view state with the proper data within

Jira

- [AAG-7](https://distillery.atlassian.net/browse/AAG-7)

**Type of change:**
- New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
- built and launched a project
- Created tests `DeleteTaskUseCaseTest`, `GetToDoListUseCaseTest`, `SaveTaskUseCasteTest`

**Checklist:**
- [x] This PR follows [guidelines](https://github.com/DistilleryTech/android-arch-blueprints/wiki/Pull-Request-Process)
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/GetToDoListUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        single<ToDoRepository> { mockRepo }
+                    }
+            )
+        }
+        MockitoAnnotations.initMocks(this)
+    }
+
+    @Test
+    fun `retrieve DataState with ToDoList`() {
+        runBlocking {
```
Comment (distillery-alexander-ageychenko) [inline mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/GetToDoListUseCaseTest.kt:45]: Minor:
```suggestion
    fun `retrieve DataState with ToDoList`() = runBlocking {
```

File: mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/GetToDoListUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    @Before
+    fun setUp() {
+        startKoin {
+            modules(
+                    module {
+                        single<ToDoRepository> { mockRepo }
+                    }
+            )
+        }
+        MockitoAnnotations.initMocks(this)
```
Comment (distillery-alexander-ageychenko) [inline mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/GetToDoListUseCaseTest.kt:40]: Minor: If I'm not mistaken we can use Nhaarman's library and ease usage of mocks https://github.com/nhaarman/mockito-kotlin

File: mvi_example/src/main/java/com/distillery/android/blueprints/mvi/todo/usecases/SaveTaskUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import com.distillery.android.blueprints.mvi.todo.state.TodoState
+import com.distillery.android.domain.ToDoRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+import java.net.ConnectException
+
+class SaveTaskUseCase : KoinComponent {
+    private val toDoRepo: ToDoRepository by inject()
```
Comment (distillery-alexander-ageychenko) [inline mvi_example/src/main/java/com/distillery/android/blueprints/mvi/todo/usecases/SaveTaskUseCase.kt:12]: Minor: it's better for readability to use full name, it's not that long so we don't ready save any time/space by cutting ending, but that increases cognitive load when reading such variables, as we read code much often then write it - it is better to concentrate on ease of readability

File: mvi_example/src/main/java/com/distillery/android/blueprints/mvi/todo/usecases/GetToDoListUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+class GetToDoListUseCase : KoinComponent {
+    private val toDoRepo: ToDoRepository by inject()
+
+    suspend fun getToDoList(): Flow<TodoState<List<ToDoModel>>> {
+        return flow {
+            toDoRepo.fetchToDos().catch { e ->
+                emit(TodoState.ErrorState(e.cause))
+            }.map { emit(TodoState.DataState(it)) }
+        }
```
Comment (distillery-alexander-ageychenko) [inline mvi_example/src/main/java/com/distillery/android/blueprints/mvi/todo/usecases/GetToDoListUseCase.kt:21]: minor: we can improve readability here
```suggestion
        return flow {
            toDoRepo.fetchToDos()
            .catch { e -> emit(TodoState.ErrorState(e.cause)) }
            .map { emit(TodoState.DataState(it)) }
        }
```

File: mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/SaveTaskUseCasteTest.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @After
+    fun after() {
+        stopKoin()
+    }
+
+    companion object {
+        private const val STRING_STUB = "non empty string"
+    }
```
Comment (distillery-alexander-ageychenko) [inline mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/SaveTaskUseCasteTest.kt:69]: minor, you can put it on top of test class, JFYI
```suggestion

private const val STRING_STUB = "non empty string"

class SaveTaskUseCasteTest : KoinTest {
...
}
```

Comment (cnavarrete101) [reply mvi_example/src/test/java/com/distillery/android/blueprints/mvi/feature/todo/SaveTaskUseCasteTest.kt:69]: that was something I didn't know!!! I learned something 😄

**Pull request merged**